KicadModTree/nodes/base/Pad.py

```diff
@@ -210,22 +316,12 @@ def _initLayers(self, **kwargs):
         self.layers = kwargs.get('layers')
 
     def _initRadiusRatio(self, **kwargs):
-        radius_ratio = kwargs.get('radius_ratio', 0)
-        if type(radius_ratio) not in [int, float]:
-            raise TypeError('radius ratio needs to be of type int or float')
-        if radius_ratio >= 0 and radius_ratio <= 0.5:
-            self.radius_ratio = radius_ratio
+        if('round_radius_handler' in kwargs):
+            self.round_radius_handler = kwargs['round_radius_handler']
         else:
-            raise ValueError('radius ratio out of allowed range (0 <= rr <= 0.5)')
-
-        if kwargs.get('maximum_radius') is not None:
-            maximum_radius = kwargs.get('maximum_radius')
-            if type(maximum_radius) not in [int, float]:
-                raise TypeError('maximum radius needs to be of type int or float')
+            self.round_radius_handler = RoundRadiusHandler(**kwargs)
 
-            shortest_sidelength = min(self.size)
-            if self.radius_ratio*shortest_sidelength > maximum_radius:
-                self.radius_ratio = maximum_radius/shortest_sidelength
+        self.radius_ratio = self.round_radius_handler.getRadiusRatio(min(self.size))
```

Comment:
Why is `min()` called with just a single argument? This happens several times in the changed code. I thought I recalled seeing a comment like this earlier, but I don't see it nor so it must have been for a different PR.

Reply:
I spotted the same thing, not sure if I submited the same question and then delted it.
But then I did some checking and learned that calling min() also works with an interable object. It returns the smallest element.

Reply:
self.size is an itterable. So min is called to get the minimum of the itterable

Reply:
Ahh. I didn't realize self.size was an iterable. Got it now.